core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java

```diff
@@ -522,15 +522,27 @@ protected DataBuffer getDataBuffer(BlockReadRequestContext context, long offset,
         blockReader = context.getBlockReader();
         Preconditions.checkState(blockReader != null);
         startTransferMs = System.currentTimeMillis();
+        ByteBuf buf;
         switch (mBlockStoreType) {
           case PAGE:
-            ByteBuffer pooledBuf = blockReader.read(offset, len);
-            return new PooledNioDataBuffer(pooledBuf, pooledBuf.remaining());
+            if (mIsReaderBufferPooled) {
+              buf = PooledNioByteBuf.allocate(len);
+            } else {
+              buf = Unpooled.directBuffer(len, len);
+            }
+            buf.clear();
```

Comment:
redundant clear() call?

Reply:
good catch, this is leftover from manual debugging.